tests/unit/test_utils_parallel.py

```diff
@@ -0,0 +1,66 @@
+"""Test multiprocessing/multithreading higher-order functions."""
+
+from importlib import import_module
+from math import factorial
+from sys import modules
+
+from pip._vendor.six import PY2
+from pip._vendor.six.moves import map
+from pytest import mark
+
+DUNDER_IMPORT = '__builtin__.__import__' if PY2 else 'builtins.__import__'
+FUNC, ITERABLE = factorial, range(42)
+MAPS = 'map_multiprocess', 'map_multithread'
+_import = __import__
+
+
+def reload_parallel():
```

Comment:
Ideally this should be called one last time after all the tests have run (maybe via an auto-use fixture at module scope) to make sure it is loaded without being affected by the tests monkeypatch.